[memgraph-v2-13 < 365] Add accumulated path and weight description (#365)
memgraph/documentation: pages/advanced-algorithms/built-in-graph-algorithms.mdx

Depth-first expansion allows an arbitrary expression filter that determines if
an expansion is allowed over a certain relationship to a certain node. The
filter is defined as a lambda function over `r` and `n`, which denotes the
relationship expanded over and node expanded to in the depth-first search.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false` and to nodes with a `drinks_USD` property less than `15`:

```cypher
MATCH path=(n {id: 0})-[* (r, n | r.eu_border = false AND n.drinks_USD < 15)]->(m {id: 8})
RETURN path;
```

> Additionally, the filter can use current path of traversal. To use it the
> filter is defined as a lambda function over `r`, `n` and `p`, which denotes the
> relationship expanded over, node expanded to and already collected path in the depth-first search.
> 
> In the following example, expansion is allowed over relationships with an `eu_border`
> property equal to `false`, to nodes with a `drinks_USD` property less than `15` and relationship type not equal to `CloseTo`:

Reviewer: ```suggestion
Additionally, the filter can use the current path of traversal. The
filter is defined as a lambda function over `r`, `n`, and `p`, which denotes the
relationship expanded over, node expanded to and the path already collected in the depth-first search.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false`, to nodes with a `drinks_USD` property less than `15` and relationship type not equal to `CloseTo`:
```

Breadth-first expansion allows an arbitrary expression filter that determines if
an expansion is allowed over a certain relationship to a certain node. The
filter is defined as a lambda function over `r` and `n`, which denotes the
relationship expanded over and node expanded to in the breadth-first search.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false` and to nodes with a `drinks_USD` property less than `15`:

```cypher
MATCH path=(n {id: 0})-[*BFS (r, n | r.eu_border = false AND n.drinks_USD < 15)]-(m {id: 8})
RETURN path;
```

> Additionally, the filter can use current path of traversal. To use it the
> filter is defined as a lambda function over `r`, `n` and `p`, which denotes the
> relationship expanded over, node expanded to and already collected path in the breadth-first search.
> 
> In the following example, expansion is allowed over relationships with an `eu_border`
> property equal to `false`, to nodes with a `drinks_USD` property less than `15` and relationship type not equal to `CloseTo`:

Reviewer: ```suggestion
Additionally, the filter can use the current path of traversal. The
filter is defined as a lambda function over `r`, `n`, and `p`, which denotes the
relationship expanded over, node expanded to, and the path already collected in the breadth-first search.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false`, to nodes with a `drinks_USD` property less than `15` and relationship type not equal to `CloseTo`:
```

RETURN path,total_weight;
```

> Additionally, the filter can use current path of traversal and current weight of path. To use them the
> filter is defined as a lambda function over `r`, `n`, `p` and `w`, which denotes the
> relationship expanded over, node expanded to, already collected path and current weight of the path in the weighted shortest path.
> 
> In the following example, expansion is allowed over relationships with an `eu_border`
> property equal to `false`, to nodes with a `drinks_USD` property less than `15`, relationship type not equal to `CloseTo` and weight less than `1000`:

Reviewer: ```suggestion
Additionally, the filter can use the current path of traversal and the current weight of the path. The
filter is then defined as a lambda function over `r`, `n`, `p`, and `w`, which denotes the
relationship expanded over, the node expanded to, the already collected path and the current weight of the path in the weighted shortest path.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false`, to nodes with a `drinks_USD` property less than `15`, relationship type not equal to `CloseTo` and weight less than `1000`:
```

```cypher
MATCH path=(n {id: 0})-[relationships:CloseTo *ALLSHORTEST (r, n | n.total_USD)]-(m {id: 51})
UNWIND (relationships(path)) AS edge
RETURN DISTINCT edge;
```

To get the total weight, add a variable at the end of the expansion expression:
```cypher
MATCH path=(n {id: 0})-[relationships:CloseTo *ALLSHORTEST (r, n | 1) total_weight]-(m {id: 9})
RETURN nodes(path), total_weight;
```

> Remember that in the case when weight is taken from the node property and edge property, `coalesce` function must be used with edge property. It's required because when Memgraph calculates weight for the first node, there is no edge in calculation.
> For example,

Reviewer: ```suggestion
When the weight is taken from the node property and relationship property, the `coalesce()` function must be used with the relationship property. It's required because when Memgraph calculates the weight for the first node, there is no relationship in the calculation.
For example:
```

> For example:

Reviewer: ```suggestion

For example:
```

MATCH path=(n {id: 0})-[*ALLSHORTEST (r, n | n.total_USD) total_weight (r, n | r.eu_border = false AND n.drinks_USD < 20)]-(m {id: 46})
RETURN path, total_weight;
```

Additionally, the filter can use the current path of traversal and the current weight of the path. In that case, the
filter is defined as a lambda function over `r`, `n`, `p`, and `w`, which denotes the relationship expanded over, node expanded to, already collected path and current weight of the path in the all shortest path.

> In the following example, expansion is allowed over relationships with an `eu_border`
> property equal to `false`, to nodes with a `drinks_USD` property less than `20`, relationship type not equal to `CloseTo` and weight less than `1000`:

Reviewer: ```suggestion
In the following example, expansion is allowed over relationships with an `eu_border` property equal to `false`, to nodes with a `drinks_USD` property less than `20`, relationship type not equal to `CloseTo` and weight less than `1000`:
```

MATCH path=(n {id: 0})-[*ALLSHORTEST (r, n | n.total_USD) total_weight (r, n | r.eu_border = false AND n.drinks_USD < 20)]-(m {id: 46})
RETURN path, total_weight;
```

> Additionally, the filter can use current path of traversal and current weight of path. To use them the
> filter is defined as a lambda function over `r`, `n`, `p` and `w`, which denotes the
> relationship expanded over, node expanded to, already collected path and current weight of the path in the all shortest path.
> 
> In the following example, expansion is allowed over relationships with an `eu_border`
> property equal to `false`, to nodes with a `drinks_USD` property less than `20`, relationship type not equal to `CloseTo` and weight less than `1000`:

Reviewer: ```suggestion
Additionally, the filter can use the current path of traversal and the current weight of the path. In that case, the
filter is defined as a lambda function over `r`, `n`, `p`, and `w`, which denotes the relationship expanded over, node expanded to, already collected path and current weight of the path in the all shortest path.

In the following example, expansion is allowed over relationships with an `eu_border`
property equal to `false`, to nodes with a `drinks_USD` property less than `20`, relationship type not equal to `CloseTo` and weight less than `1000`:
```
